### ISSUE TITLE
Fix CLI printing error information to STDOUT

### ISSUE THREAD

**Issue by naokikimura** [stylelint/stylelint#4798]
> Clearly describe the bug

CLI printed error information to STDOUT. This is bad behavior for CLI.

This problem is tricky. If stylelint encounters an error while using the pipeline, the error information may not be output.

> Which rule, if any, is the bug related to?

N/A

> What code is needed to reproduce the bug?

N/A

> What stylelint configuration is needed to reproduce the bug?

N/A

> Which version of stylelint are you using?

`13.5.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI
```sh
stylelint --print-config /dev/null | jq
```

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

N/A

> What did you expect to happen?

The CLI is expected to print error information to standard error.

> What actually happened (e.g. what warnings or errors did you get)?

Error information is printed to standard output.
